c2corg_ui/static/js/searchFilters.js

```diff
@@ -0,0 +1,127 @@
+ * 
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ */
+app.searchFiltersDirective = function() {
+  return {
+    restrict: 'A',
+    controller: 'appSearchFiltersController',
+    bindToController: true,
+    scope: true,
+    controllerAs: 'searchFiltersCtrl',
+    link: function(scope, element, attrs) {
+      // Init sliders
+      $('.range-between').slider({min: 0, max: 9999, value: [0, 9999], focus: true, tooltip: 'hide'});
+      var elevationFilter = $('#elevation-filter');
+      var heightFilter = $('#height-filter');
+
+    // Update elevation slider values
+      elevationFilter.on('slide', function(slideEvt) {
+        scope.elevation.min = slideEvt.value[0];
+        scope.elevation.max = slideEvt.value[1];
+        scope.$apply();
+      });
+
+    // Update height diff slider values
+      heightFilter.on('slide', function(slideEvt) {
+        scope.height_diff_up.min = slideEvt.value[0];
+        scope.height_diff_up.max = slideEvt.value[1];
+        scope.$apply();
+      });
+    }
+  }
+};
+
+
+app.module.directive('appSearchFilters', app.searchFiltersDirective);
+
+
+/**
+ * @param {angular.Scope} $scope Scope.
+ * @param {angular.JQLite} $element Element.
+ * @param {angular.Attributes} $attrs Attributes.
+ * @constructor
+ * @ngInject
+ * @export
+ */
+app.SearchFiltersController = function($scope, $element, $attrs) {
+  
+  /**
+   * @type {angular.Scope}
+   * @private
+   */
+  this.scope_ = $scope;
+
+  // Init filters object
+  angular.extend(this.scope_,  {
+    activities : [], waypoint_types : [], culture : null, around : null, 
+    height_diff_up : {
+      min: 0, max: 9999
+    },
+    elevation : {
+      min : 0, max: 9999
+    }
+  });
+};
+ 
+ 
+ /**
+ * @param {string} option name
+ * @param {string} optionCategory category
+ * @param {Object} e event
+ * @ngInject
+ * @export
+ */
+app.SearchFiltersController.prototype.selectOption = function(option, optionCategory, e) {
+  // Don't close the menu after selecting an option
+  e.stopPropagation();
+
+  // Shorthand
+  var scopeOptionCat = this.scope_[optionCategory];
+  var target = $(e.currentTarget);
+  var checkbox = target.find('input');
+  
+  // If this property doesn't exit in the scope, create an array and push the selected option.
+  if (!scopeOptionCat) {
+    this.scope_[optionCategory] = [];
+    updateFilterArray(this.scope_[optionCategory], target, checkbox, option);
+  }else {
+    updateFilterArray(scopeOptionCat, target, checkbox,option);
+  }
+  /**
+ * @param {Array} scopeOptionCat 
+ * @param {Object} target
+ * @param {Object}checkbox
+ * @param {string}option
+ * Push/remove the selected option into the scope property (scope_[optionCat]) and
+ * check/uncheck the checkbox depending if the options is already in the array or not.
+ */
+  function updateFilterArray(scopeOptionCat, target, checkbox,option) {
+    if ($.inArray(option,  scopeOptionCat) === -1) {
```

Comment:
extra space before ``scopeOptionCat``.
In addition I think you should use the stand javascript ``indexOf`` function instead of ``$.inArray``. @gberaudo what do you think?